validator/rpc/auth.go

```diff
@@ -22,30 +22,34 @@ import (
 )
 
 var (
-	tokenExpiryLength = 20 * time.Minute
+	tokenExpiryLength = time.Hour
 	hashCost          = 8
 )
 
 // Signup to authenticate access to the validator RPC API using bcrypt and
 // a sufficiently strong password check.
 func (s *Server) Signup(ctx context.Context, req *pb.AuthRequest) (*pb.AuthResponse, error) {
+	walletDir := s.walletDir
+	if req.WalletDir != "" {
```

Comment:
Maybe trim the value before comparison to disallow dirs containing whitespace only.